Update menu.go

600ML opened the pull request:
json: cannot unmarshal string into Go struct field MatchRule.conditionalmenu.matchrule.group_id of type int32
接口返回是字符串
"matchrule": {
"group_id": "2"
}

结构体定义
//MatchRule 个性化菜单规则
type MatchRule struct {
GroupID int32 json:"group_id,omitempty"
...
}

600ML committed:
Update menu.go

json: cannot unmarshal string into Go struct field MatchRule.conditionalmenu.matchrule.group_id of type int32
接口返回是字符串
"matchrule": {
"group_id": "2"
}

结构体定义
//MatchRule 个性化菜单规则
type MatchRule struct {
GroupID int32 json:"group_id,omitempty"
...
}

silenceper commented:
多谢

silenceper commented:
`sex`, `client_platform_type`字段是不是需要统一都改成`string`

silenceper commented on officialaccount/menu/menu.go:
```diff
@@ -102,7 +102,7 @@ type ButtonNew struct {
 type MatchRule struct {
-	GroupID            int32  `json:"group_id,omitempty"`
+	GroupID            string  `json:"group_id,omitempty"`
 	Sex                int32  `json:"sex,omitempty"`
```
这里int32的字段是不是都需要统一改成`string`

silenceper reviewed (changes requested):
这里int32的字段是不是都需要统一改成string

600ML committed:
matchrule字段均是字符串

 "matchrule": {
        "tag_id": "2", 
        "sex": "1", 
        "country": "中国", 
        "province": "广东", 
        "city": "广州", 
        "client_platform_type": "2", 
        "language": "zh_CN"
    }

xinzf commented:
MatchRule 里面缺少tag_id
